python/pyarrow/types.pxi

```diff
@@ -564,22 +574,30 @@ cdef class Schema:
                 self.schema.metadata())
             return box_metadata(metadata.get())
 
-    def __richcmp__(self, other, int op):
-        if op == cp.Py_EQ:
+    def __eq__(self, other):
```

Comment:
Is this the best practice for extension types? I thought it was to use `__richcmp__`

Reply:
Since Cython 0.27 there is no preference anymore, one can choose freely. Before 0.27 you could only use `__richcmp__`, `__eq__` was not possible.

Reply:
I think `__eq__` is more idiomatic, also this way we can compare against arbitrary object without raising a TypeError.

Reply:
:+1: I didn't know that Cython 0.27 brought this improvement